### PR TITLE
SONAR-7986 Create /api/settings/reset WS to remove the value of a setting

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-7986 Create /api/settings/reset WS to remove the value of a setting

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/settings/ws/ResetAction.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  private Optional<ComponentDto> getComponent(DbSession dbSession, ResetRequest request) {
+    if (request.getComponentId() == null && request.getComponentKey() == null) {
+      return Optional.empty();
+    }
+    ComponentDto project = componentFinder.getByUuidOrKey(dbSession, request.getComponentId(), request.getComponentKey(), ComponentFinder.ParamNames.COMPONENT_ID_AND_KEY);
+    return Optional.of(project);
+  }
+
+  private void checkPermissions(Optional<ComponentDto> component) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ResetAction.java]: MINOR Specify a "ComponentDto" parameter instead. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3553)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: server/sonar-server/src/main/java/org/sonar/server/settings/ws/ResetAction.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      .setRequired(true);
+    addComponentParameters(action);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      ResetRequest resetRequest = toWsRequest(request);
+      Optional<ComponentDto> component = getComponent(dbSession, resetRequest);
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ResetAction.java]: minor: if the result is Optional, then I would name it `searchComponent`. I'd use `getComponent` when the result is found or else it fails

File: server/sonar-server/src/main/java/org/sonar/server/settings/ws/ResetAction.java (hunk 1 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/settings/ws/ResetAction.java]: @benzonico I think Optional should be allowed as a parameter in **private** methods.

**Comment by teryk** [comment]
LGTM

**Pull request merged**